apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -1007,4 +1009,49 @@ export class ApplicationController {
 
     return { url }
   }
+
+  @Get('applications/:id/attachments/:s3Key/presigned-url')
+  @Scopes(ApplicationScope.read)
+  @Documentation({
+    description: 'Gets a presigned url for attachments',
+    response: { status: 200, type: PresignedUrlResponseDto },
+    request: {
+      query: {},
+      params: {
+        id: {
+          type: 'string',
+          description: 'application id',
+          required: true,
+        },
+        s3Key: {
+          type: 'string',
+          description: 's3 key for attachment',
+          required: true,
+        },
+      },
+    },
+  })
+  async getAttachmentPresignedURL(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @Param('s3Key') s3Key: string,
```

Comment:
Let's call this `attachmentKey` instead. S3 is an implementation detail.

Reply:
yep - had been brewing in my mind - good call